pkg/monitortests/machines/watchmachines/machines.go

```diff
@@ -0,0 +1,68 @@
+package watchmachines
+
+import (
+	"context"
+	"time"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+
+	machine "github.com/openshift/api/machine/v1beta1"
+	machineClient "github.com/openshift/client-go/machine/clientset/versioned"
+	machineinformerv1 "github.com/openshift/client-go/machine/informers/externalversions/machine/v1beta1"
+	"k8s.io/client-go/tools/cache"
+)
+
+func startMachineMonitoring(ctx context.Context, m monitorapi.RecorderWriter, client machineClient.Interface) {
+
+	machinePhaseChangeFns := []func(machine, oldMachine *machine.Machine) []monitorapi.Interval{
+		func(machine, oldMachine *machine.Machine) []monitorapi.Interval {
+			var intervals []monitorapi.Interval
+			now := time.Now()
+			if machine.Status.Phase != nil && oldMachine.Status.Phase != nil && *machine.Status.Phase != *oldMachine.Status.Phase {
+				intervals = append(intervals,
+					monitorapi.NewInterval(monitorapi.SourceMachine, monitorapi.Warning).
+						Locator(monitorapi.NewLocator().MachineFromName(machine.Name)).
+						Message(monitorapi.NewMessage().Reason(monitorapi.MachinePhaseChanged).
+							HumanMessage("changed")).
+						Build(now, now))
```

Comment:
Does this mean it's an instantaneous point? Since the phases are linear, I wonder how we can represent the phase being set for a particular period

Or does this initiate the phase and end only when there's an alternative value set?

Reply:
Yea I'm not sure how to actually get an interval in this case. There was a StateTrackker but that seems to rely on events. I'm not sure if machine phase changes triggers an event.

Reply:
As far as I can tell, no, it doesn't trigger an event at the moment